Source/DataSources/StaticGroundGeometryPerMaterialBatch.js

```diff
@@ -224,133 +254,117 @@ define([
             return BoundingSphereState.FAILED;
         }
-
-        bs.clone(result);
+        attributes.boundingSphere.clone(result);
         return BoundingSphereState.DONE;
     };
 
-    Batch.prototype.removeAllPrimitives = function() {
-        var primitives = this.primitives;
-
+    Batch.prototype.destroy = function() {
         var primitive = this.primitive;
+        var primitives = this.primitives;
         if (defined(primitive)) {
             primitives.remove(primitive);
-            this.primitive = undefined;
-            this.geometry.removeAll();
-            this.updaters.removeAll();
         }
-
         var oldPrimitive = this.oldPrimitive;
         if (defined(oldPrimitive)) {
             primitives.remove(oldPrimitive);
-            this.oldPrimitive = undefined;
         }
+        this.removeMaterialSubscription();
     };
 
     /**
      * @private
      */
-    function StaticGroundGeometryColorBatch(primitives, classificationType) {
-        this._batches = new AssociativeArray();
+    function StaticGroundGeometryPerMaterialBatch(primitives, appearanceType, classificationType) {
+        this._items = [];
         this._primitives = primitives;
+        this._appearanceType = appearanceType;
         this._classificationType = classificationType;
     }
 
-    StaticGroundGeometryColorBatch.prototype.add = function(time, updater) {
-        var instance = updater.createFillGeometryInstance(time);
-        var batches = this._batches;
-        // instance.attributes.color.value is a Uint8Array, so just read it as a Uint32 and make that the key
-        var batchKey = new Uint32Array(instance.attributes.color.value.buffer)[0];
-        var batch;
-        if (batches.contains(batchKey)) {
-            batch = batches.get(batchKey);
-        } else {
-            batch = new Batch(this._primitives, this._classificationType, instance.attributes.color.value, batchKey);
-            batches.set(batchKey, batch);
+    StaticGroundGeometryPerMaterialBatch.prototype.add = function(time, updater) {
+        var items = this._items;
+        var length = items.length;
+        var geometryInstance = updater.createFillGeometryInstance(time);
+        var usingSphericalTextureCoordinates = ShadowVolumeAppearance.shouldUseSphericalCoordinates(geometryInstance.geometry.rectangle);
+        for (var i = 0; i < length; ++i) {
+            var item = items[i];
+            if (item.isMaterial(updater) &&
+                item.nonOverlapping(geometryInstance.geometry.rectangle) &&
+                item.usingSphericalTextureCoordinates === usingSphericalTextureCoordinates) {
+                item.add(time, updater, geometryInstance);
+                return;
+            }
         }
-        batch.add(updater, instance);
-        return batch;
+        var batch = new Batch(this._primitives, this._appearanceType, this._classificationType, updater.fillMaterialProperty, usingSphericalTextureCoordinates);
```

Comment:
This appears to create a new batch every time we create a new entity, so nothing is being batched together.  That can't be what we want

Reply:
I'll add some doc. Realized the "batchability" rules might also seem weird without an explanation.